opentimelineio_contrib/adapters/xges.py

```diff
@@ -0,0 +1,774 @@
+        self.values[key] = value
+
+    def get(self, key, default=None):
+        return self.values.get(key, default)
+
+    @staticmethod
+    def _find_eos(s):
+        # find next '"' without preceeding '\'
+        l = 0
+        while 1:  # faster than regexp for '[^\\]\"'
+            p = s.index('"')
+            l += p + 1
+            if s[p - 1] != '\\':
+                return l
+            s = s[(p + 1):]
+        return -1
+
+    @staticmethod
+    def escape(s):
+        if not isinstance(s, str):
+            return s
+        return s.replace(" ", "\\ ")
+
+    @staticmethod
+    def _parse(s):
+        in_string = s
+        types = {}
+        values = {}
+        scan = True
+        # parse id
+        p = s.find(',')
+        if p == -1:
+            try:
+                p = s.index(';')
+            except ValueError:
+                p = len(s)
+            scan = False
+        name = s[:p]
+        # parse fields
+        while scan:
+            comma_space_it = p
+            # skip 'name, ' / 'value, '
+            while s[comma_space_it] in [' ', ',']:
+                comma_space_it += 1
+            s = s[comma_space_it:]
+            p = s.index('=')
+            k = s[:p]
+            if not s[p + 1] == '(':
+                raise ValueError("In %s position: %d" % (in_string, p))
+            s = s[(p + 2):]  # skip 'key=('
+            p = s.index(')')
+            t = s[:p]
+            s = s[(p + 1):]  # skip 'type)'
+
+            if s[0] == '"':
+                s = s[1:]  # skip '"'
+                p = GstStructure._find_eos(s)
+                if p == -1:
+                    raise ValueError
+                v = s[:(p - 1)]
+                if s[p] == ';':
+                    scan = False
+                # unescape \., but not \\. (using a backref)
+                # need a reverse for re.escape()
+                v = v.replace('\\\\', '\\')
+                v = GstStructure.UNESCAPE.sub(r'\1', v)
+            else:
+                p = s.find(',')
+                if p == -1:
+                    p = s.index(';')
+                    scan = False
+                v = s[:p]
+
+            if t == 'structure':
+                v = GstStructure(v)
+            elif t == 'string' and len(v) and v[0] == '"':
+                v = v[1:-1]
+            elif t == 'boolean':
+                v = (v == '1')
+            elif t in GstStructure.INT_TYPES:
+                v = int(v)
+            types[k] = t
+            values[k] = v
+
+        return (name, types, values)
+
+
+class GESTrackType:
+    UNKNOWN = 1 << 0
+    AUDIO = 1 << 1
+    VIDEO = 1 << 2
+    TEXT = 1 << 3
+    CUSTOM = 1 << 4
+
+    @staticmethod
+    def to_otio_type(_type):
+        if _type == GESTrackType.AUDIO:
+            return otio.schema.TrackKind.Audio
+        elif _type == GESTrackType.VIDEO:
+            return otio.schema.TrackKind.Video
+
+        assert(("Can't translate track type %s" % _type) is None)
```

Comment:
To match the error reporting pattern used elsewhere in OTIO, we recommend making a subclass of Exception, and using raise instead of assert.

Something like this:
```
class GSTParseError(otio.exceptions.OTIOError):
    pass
```
and then
`raise GSTParseError("Can't translate track type %s" % _type)`

Reply:
Fixed.